### DISCUSSION TITLE
use customed fontfile for insertTextbox/insertText?

### DISCUSSION THREAD

**Discussion by retsyo** [pymupdf/PyMuPDF#1669] [Looking for help]
in the following code, a chinese ttf font file is used. however only `TextWriter` treats it correctly, `insertTextbox/insertText` fails.
Any solution? Thanks

BTW, I have read https://github.com/pymupdf/PyMuPDF/issues/329

```python
import fitz
doc = fitz.open()
page = doc.newPage()

text = "你好1234测试"

fontfile = r'C:\Windows\Fonts\simsunb.ttf'
font = fitz.Font(fontfile=fontfile)

tw = fitz.TextWriter(page.rect)
tw.append( (100, 50), '0 - ' + text, font=font) # is ok
tw.write_text(page)

page.insertText((100,100), '1 - ' + text, fontname="china-s") # is ok

page.insertText((100,150), '2 - ' + text, fontfile = fontfile)  # bad

page.insertTextbox((100,200,320,260), '3 - ' + text, fontname="china-ss") # is ok

page.insertTextbox((100,250,320,310), '4 - ' + text, fontfile = fontfile) # bad

doc.save("test.pdf")

```

![mathml](https://user-images.githubusercontent.com/7960913/162619432-290403f1-cf80-46f1-aeca-ae0fd59976a0.png)

**Comment by JorjMcKie** [comment]
The recommended way using font with ``insert_text(box)`` is th first insert the font on the page, and then - independently - using that font in hte text insertions:

```python
page.insert_font(..., fontname="xyz", fontfile="...")  # or fontbuffer=
...
page.insert_text(pos,text,fontname="xyz")
```
The fontname ("reference name") must be unique and not equal to any of the Base-14 names.
